documentation/modules/ROOT/pages/connectors/jdbc.adoc

```diff
@@ -0,0 +1,660 @@
+=== Primary key handling
+
+By default, the {prodname} JDBC sink connector does not associate any fields from the source event to a primary key.
+Unfortunately, this is not always a good choice depending on the business requirements and configuration of the sink connector, such as using upsert semantics.
+The connector can be configured with several different primary key modes:
+
+|===
+|Mode|Description
+
+|`none`
+|No primary key fields are specified when creating the table.
+
+|`kafka`
+|The primary key will consist of three columns `\__connect_topic`, `__connect_partition`, and `__connect_offset`.
+The values for these columns will be sourced from the Kafka event's coordinates.
+
+|`record_key`
+|The primary key will be composed of the Kafka event's key. +
+ +
+If the primary key is a primitive type, the `primary.key.fields` property must be specified, which specifies the name of the column to be used.
+If the primary key is a struct type, the fields in the struct will be mapped as columns of the primary key.  The `primary.key.fields` property can be used to restrict the primary key to a subset of columns.
+
+|`record_value`
+|The primary key will be composed of the Kafka event's value. +
+ +
+Since the Kafka event's value must always be a `Struct`, all fields within the value will be used as primary key columns by default.
+In order to use a subset of fields, specify the `primary.key.fields` property with a comma-separated list of fields from the value to derive the primary key columns of.
+
+|===
+
+[IMPORTANT]
+====
+Some dialects may throw an exception if you set the `primary.key.mode` to `kafka` and set `schema.evolution` to `basic`.
+This exception occurs because some dialects map a `STRING` data type mapping to a variable length string data type such as `TEXT` or `CLOB`, and these dialects do not allow unbounded length primary key columns.
+To avoid this problem, the following must be set:
+
+* Do not set `schema.evolution` to `basic`.
+* Create the database table and primary key mappings in advance.
+====
+
+[[jdbc-delete-mode]]
+=== Delete mode
+
+The {prodname} JDBC sink connector can delete rows in the destination database when a `DELETE` or _tombstone_ event is consumed.
+By default, the JDBC sink connector does not enable delete mode, therefore, it must be enabled explicitly in the connector configuration if you wish to support removing rows.
+
+Deletes can be enabled by setting `delete.enabled=true` in the connector configuration, but this mode requires that the `primary.key.mode` be set to a value that is not `none`.
+This is because deletes are executed based on the primary key mapping, and therefore is a destination table has no primary key mapping, rows cannot be deleted by the connector.
+
+[[jdbc-idempotent-writes]]
+=== Idempotent writes
+
+The {prodname} JDBC sink connector supports idempotent writes, allowing the same records to be replayed repeatedly and the final database state remains consistent.
```

Comment:
```suggestion
The {prodname} JDBC sink connector supports idempotent writes, so that even if the same records are applied multiple times, the final database state is not changed after the first application.
```
Ok, I thnk I get it. Does this ^ work?